### PR TITLE
Fix/cors middleware

### PR DESCRIPTION
**Pull request by opaduchak**
Fixed not being able to call GV from frontend due to missing CORS headers

### CODE REVIEW

**Review by aaxelb — COMMENTED** [review]
looks good! would you mind rebasing your changes onto `develop`? (i've now merged `assorted-betterment`, and you shouldn't need the migration changes)

File: addon_service/migrations/0002_alter_externalstorageservice_int_addon_imp_and_more.py
Comment (aaxelb) [inline addon_service/migrations/0002_alter_externalstorageservice_int_addon_imp_and_more.py:1]: you should be able to delete this (the `verbose_name` changes are now reflected in `0001_initial`

**Review by aaxelb — CHANGES_REQUESTED** [review]
remove the changes to migrations and it's good to go, thanks!

File: app/env.py [hunk 1]
```diff
@@ -31,3 +31,4 @@
 
 # comma-separated list:
 ALLOWED_HOSTS = os.environ.get("ALLOWED_HOSTS", "").split(",")
+CORS_ALLOWED_ORIGINS = os.environ.get("CORS_ALLOWED_ORIGINS", "").split(",")
```
Comment (aaxelb) [inline app/env.py:34]: when CORS_ALLOWED_ORIGINS isn't set it defaults to `[""]`, which [causes an error](https://github.com/CenterForOpenScience/gravyvalet/actions/runs/9598391564/job/26479890621?pr=69#step:8:23) 
```suggestion
CORS_ALLOWED_ORIGINS = tuple(filter(bool, os.environ.get("CORS_ALLOWED_ORIGINS", "").split(",")))
```
(i guess ALLOWED_HOSTS is not validated as strictly, but could do with the same fix)

**Review by aaxelb — CHANGES_REQUESTED** [review]
small bug from the CI run

File: app/env.py (hunk 1 above)
Comment (opaduchak) [reply app/env.py]: done

File: addon_service/migrations/0002_alter_externalstorageservice_int_addon_imp_and_more.py
Comment (opaduchak) [reply addon_service/migrations/0002_alter_externalstorageservice_int_addon_imp_and_more.py:1]: done

**Review by aaxelb — APPROVED** [review]
👍

**Comment by opaduchak** [comment]
So, can I merge this now @aaxelb?

**Pull request merged**